### PR TITLE
BaSP-TG-63: Admins Shared component implementation & redux

### PR DESCRIPTION
**Pull request by lucianofrailelopez**
With the shared components, redo Admins screen so it doesn't redirect to other routes and then implement Redux.

Add and Edit forms should be inside modals. Also all alert/success messages should be in modals.

Shared components used:

Preloader
Table
Add super admin modal
Edit super admin modal
Delete confirmation modal
Success/Error modal
Buttons
Form inputs

warning: delete's modal confirm close-icon is modified in another next PR and there is an error with error-success confirm but the add and edit function can be performed successfully.

### CODE REVIEW

**Review by NicolasFontana — CHANGES_REQUESTED** [review]
Also, you should correct the size of the add and edit modal.
image
image

File: src/Components/Admins/Edit/index.js [hunk 1]
```diff
@@ -11,100 +20,96 @@ const AdminsEdit = () => {
...
-          window.location.replace(
-            'https://alexis-trackgenix-app-git-feature-tg-28admin-basp-m2022.vercel.app/admins'
-          );
-        } else if (response.message) {
-          alert(response.message);
-        }
-      });
+  const onChange = (event) => {
+    setadminInput({ ...adminInput, [event.target.name]: event.target.value });
+    // setEdited(true);
```
Comment (agusting19) [inline src/Components/Admins/Edit/index.js:35]: This comment can be deleted

**Review by agusting19 — COMMENTED** [review]
There are some commented lines to deleted and also problems with the esthetic. As Nico show us, the modal's need changes in the css, also, when we delete an Admin we never see the success.

**Review by agusting19 — COMMENTED** [review]
I saw that you were able to correct the css of the modal but there are still commented lines of code. Please review the corrections that were made to you

**Review by NicolasFontana — APPROVED** [review]
Nice work!

**Review by agusting19 — APPROVED** [review]
Nice work buddy!

**Comment by LeonelListro** [comment]
# Testing Results

**Tested locally on:** https://alexis-trackgenix-app-git-feature-tg-63-share-0bd81f-basp-m2022.vercel.app/admins

Test result: Failed ⚠️

_Bugs found:_

1- Password should not be showed. 
modal

2- When you submit whit an error data, the loader stays on the screen forever.
modal

3- The modal that appears at first is not the correct one.
modal

4- Modal should be centered in relation to the title.
modal

## The above errors are repeated at all checkpoints.

**Review by LeonelListro — CHANGES_REQUESTED** [review]
Please refer to https://github.com/BaSP-m2022/alexis-trackgenix-app/pull/26#issuecomment-1159821736

File: src/Components/Shared/confirmationModal/confirmModal.module.css [hunk 2]
```diff
@@ -37,6 +37,8 @@
     left: 130px;
     margin-right: 10px;
     color: #ffffff;
+    background-color: transparent;
+    border: none;
```
Comment (Javier-Cervera) [inline src/Components/Shared/confirmationModal/confirmModal.module.css:41]: Modal will be modified on another PR

Comment (LeonelListro) [reply src/Components/Shared/confirmationModal/confirmModal.module.css:43]: Ok, should take in count the position of the delete modal too. (it's not centered taking in count the screen like others modals, and it should be).

**Comment by LeonelListro** [comment]
Tested localy on: https://alexis-trackgenix-app-git-feature-tg-63-share-0bd81f-basp-m2022.vercel.app/admins

Previous corrections are approved ✔

New bugs found: ⚠

1- When you try to create a new admin with an invalid data on a field the error message modal deploy correctly, but if you try to close the error message modal and the add admin modal with the cross button, the page get crashed.

image

**Review by LeonelListro — CHANGES_REQUESTED** [review]
New bugs found, please refer to https://github.com/BaSP-m2022/alexis-trackgenix-app/pull/26#issuecomment-1159918178

**Review by LeonelListro — APPROVED** [review]
Well done Luciano and team!

**Pull request merged**